src/tableHeaderCells.js

```diff
@@ -0,0 +1,67 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Table } from 'evergreen-ui'
+
+const TableSortHeaderCell = ({columnNumber, headerText, width, orderedColumnIndex, ascendingOrder, onClick}) =>
+  <Table.TextHeaderCell
+    className={`header${columnNumber}`}
+    flexBasis={width} flexShrink={100} flexGrow={100}
+    onClick={() => onClick(columnNumber)}>
+    <div style={{display: `ruby`}}>
+      {headerText}
+      { columnNumber===orderedColumnIndex ?
+        ascendingOrder ?
+          <i className={`icon icon-common icon-sort-up`} style={{paddingLeft: `1em`}}/> :
+          <i className={`icon icon-common icon-sort-down`} style={{paddingLeft: `1em`}}/> :
+        <i className={`icon icon-common icon-sort`} style={{paddingLeft: `1em`}}/>
+      }
+    </div>
+  </Table.TextHeaderCell>
+
+const TableSearchHeaderCell = ({columnNumber, headerText, width, searchedColumnIndex, searchQuery, onChange}) =>
+  <Table.SearchHeaderCell
+    className={`searchheader${columnNumber}`}
+    flexBasis={width} flexShrink={100} flexGrow={100}
+    onChange={value => onChange(value, columnNumber)}
+    value={columnNumber===searchedColumnIndex ? searchQuery : ``}
+    placeholder = {`Search by ${headerText} ...`}
+  />
+
+const tableHeaderCells = (tableHeader, searchedColumnIndex, searchQuery, orderedColumnIndex, ascendingOrder, onClick, onChange ) => {
+  return tableHeader.map((header, index) => {
+    switch(header.type) {
+    case `plain`:
+      return <Table.TextHeaderCell key={header.title} flexBasis={header.width} flexShrink={100} flexGrow={100}>
+        {header.title}
+      </Table.TextHeaderCell>
+    case `sort`:
+      return <TableSortHeaderCell  key={`sortheader${index}`} columnNumber={index} headerText={header.title} width={header.width}
+        {...{orderedColumnIndex, ascendingOrder, onClick}}/>
+    case `search`:
+      return <TableSearchHeaderCell key={`searchheader${index}`} columnNumber={index} headerText={header.title} width={header.width}
+        {...{searchedColumnIndex, searchQuery, onChange}}/>
+    default:
+      return <Table.TextHeaderCell key={header.title}>{header.title}</Table.TextHeaderCell>
```

Comment:
Do we need to consider both a `plain` type which we’re not using and the default? Couldn’t we just have `sort`, `order` and `default`?

Reply:
Yes. I think we are not going to use a `default` option different from `plain`.